src/requirements/requirement-constraints-utils.ts

```diff
@@ -53,7 +54,6 @@ export const getConstraintViolationsForSingleCourse = <Requirement extends strin
         requirement,
         constraintViolatingRequirementsForCurrentRequirement
       );
-      constraintViolationsGraph.addRequirementNode(requirement);
```

Comment:
porque?

Reply:
unnecessary/redundant -- addEdge does this as the first step